packages/block-editor/src/components/collab/collab-board/index.js

```diff
@@ -0,0 +1,442 @@
+/**
+ * WordPress dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import apiFetch from '@wordpress/api-fetch';
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import {
+	TextControl,
+	Button,
+	CheckboxControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Modal,
+} from '@wordpress/components';
+import { dateI18n, format, getSettings } from '@wordpress/date';
+import {
+	commentAuthorAvatar as userIcon,
+	Icon,
+	trash as deleteIcon,
+	edit as editIcon,
+} from '@wordpress/icons';
+
+import { useSelect, useDispatch } from '@wordpress/data';
+//import { store as coreStore } from '@wordpress/core-data';
+//import { useEntityProp } from '@wordpress/core-data';
+
+export default function BlockCommentModal( { clientId, onClose, threadId } ) {
+	// State to manage the comment thread.
+	const [ inputComment, setInputComment ] = useState( '' );
+	const [ isResolved, setIsResolved ] = useState( false );
+	const [ isEditing, setIsEditing ] = useState( null );
+	const [ showConfirmation, setShowConfirmation ] = useState( false );
+	const curruntUserData = useSelect( ( select ) => {
+		// eslint-disable-next-line @wordpress/data-no-store-string-literals
+		return select( 'core' ).getCurrentUser();
+	}, [] );
+
+	const userAvatar = curruntUserData.avatar_urls[ 48 ] || null;
+	const currentUser = curruntUserData?.name || null;
+
+	const allThreads = [];
+	const currentThread = allThreads[ threadId ] ?? {};
+	const isCurrentThreadResolved = currentThread.threadIsResolved || false;
+	const commentsCount = isCurrentThreadResolved
+		? 0
+		: currentThread.comments?.length || 0;
+	// eslint-disable-next-line @wordpress/data-no-store-string-literals
+	const postID = useSelect( ( select ) => {
+		// eslint-disable-next-line @wordpress/data-no-store-string-literals
+		return select( 'core/editor' ).getCurrentPostId();
+	}, [] );
+
+	// Get the dispatch functions to save the comment and update the block attributes.
+	// eslint-disable-next-line @wordpress/data-no-store-string-literals
+	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
+	// Helper function to generate a new comment.
+	const generateNewComment = () => ( {
+		commentId: Date.now(),
+		createdBy: currentUser,
+		comment: inputComment,
+		createdAt: new Date().toISOString(),
+	} );
+
+	// // Function to add a border class to the content reference.
+	const setAttributes = () => {
+		updateBlockAttributes( clientId, {
+			className: `block-editor-collab__${ threadId }`,
```

Comment:
I commented this before, but we shouldn't override the className attribute of a block. We should use a custom attribute and derive the class name from that.

Reply:
Done with custom attribute